### PR TITLE
Integrate MessageInputBar as a dependency

### PR DESCRIPTION
**Pull request by nathantannar4**
This PR removes all source code / test related to MessageInputBar and replaces by setting `MessageInputBar/Core` as a new dependency.  This will also for separate development of each library without bloating MessageKit through the additions such as autocomplete (available in `MessageInputBar/AutocompleteManager`

If there are any Cartfile users here please let me know if the changes I made to the Cartfiles are correct.

### CODE REVIEW

**Comment by nathantannar4** [comment]
For #730

File: Cartfile.private [hunk 1]
```diff
@@ -1,2 +1,3 @@
 github "Quick/Quick" ~> 1.2.0
 github "Quick/Nimble" ~> 7.0.3
+github "MessageKit/MessageInputBar" ~> 0.2.0
```
Comment (SD10) [inline Cartfile.private]: This would need to be declared in a `Cartfile`, not `Cartfile.private`

**Comment by SD10** [comment]
@nathantannar4 We'll also need to update the Circle CI configuration to build the dependencies. You can see how we do it at Moya

**Comment by SD10** [comment]
@nathantannar4 Want me to take over this one?

**Comment by nathantannar4** [comment]
@SD10 Ya I don't know why the build is failing, something to do with the Carthage things I think

**Comment by SD10** [comment]
If any @MessageKit/contributors want to pick this one up that would be 🔥 😊 Need to link this in a way that supports both CocoaPods and Carthage

**Comment by zhongwuzw** [comment]
Emm, what about make `MessageInputBar` as a git submodule?

**Comment by SD10** [comment]
@zhongwuzw I'm not familiar with git submodules. I'd like to set up this project the same way as something like Moya

**Comment by ndonald2** [comment]
Carthage should allow specifying other dependencies without using a submodule. My guess is that this is why it's failing (input bar dependency in the private, not public, cartfile)

https://github.com/MessageKit/MessageKit/pull/732#discussion_r195914957

**Comment by SD10** [comment]
@nathantannar4 We need to create a new file `Cartfile` and list the dependency there. The `Cartfile.private` is used for dev and `Cartfile` for release

Check out the [Cartfile](https://github.com/Moya/Moya/blob/master/Cartfile) at Moya

**Comment by nathantannar4** [comment]
@SD10 we do have a `Cartfile`

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by nathantannar4** [comment]
It seems to build it correctly now, some random test error occurred though

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
@nathantannar4 There is no Cartfile

https://github.com/MessageKit/MessageKit/blob/split-messageinputbar/Cartfile

**Comment by nathantannar4** [comment]
Works now

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Review by zhongwuzw — APPROVED** [review]
Good 👍 ,waiting for `2.0`

**Comment by SD10** [comment]
Will review this weekend. I’d like to see this make it in 2.0. We’ll have to do a beta for a change this big

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Review by SD10 — APPROVED** [review]
Ok looks like we won't be able to do a 1.1 release but this is good to go and I need it 😓

**Pull request merged**